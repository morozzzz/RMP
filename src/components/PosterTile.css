.poster-tile {
    width: 80%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 1em 1.5em;
}

.poster-image {
    width: 100%;
}

.primary-data {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 0.5em 0;
}

.poster-title {
    font-size: 0.9em;
    font-weight: bold;
}

.poster-year{
    font-size: 0.6em;
    border: 0.08em solid black;
    border-radius: 15%;
    padding: 0.2em 0.6em;
    display: flex;
}

.secondary-data {
    width: 100%;
    display: flex;
    justify-content: flex-start;
}

.poster-genre {
    font-size: 0.9em;
    color: rgb(90, 90, 90);
}