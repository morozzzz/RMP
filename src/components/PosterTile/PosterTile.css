.poster-tile {
    width: 80%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 1em 1.5em;
    cursor: pointer;
}

.poster-tile__image {
    width: 100%;
}

.poster-tile__primary-data {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 0.5em 0;
}

.poster-tile__title {
    font-size: 0.9em;
    font-weight: bold;
}

.poster-tile__year{
    font-size: 0.6em;
    border: 0.08em solid black;
    border-radius: 15%;
    padding: 0.2em 0.6em;
    display: flex;
}

.poster-tile__secondary-data {
    width: 100%;
    display: flex;
    justify-content: flex-start;
}

.poster-tile__genre {
    font-size: 0.9em;
    color: rgb(90, 90, 90);
}
