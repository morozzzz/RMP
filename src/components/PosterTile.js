import React from 'react';
import './PosterTile.css';

const PosterTile = ({ data }) => {
    const {
        imageURL,
        title,
        genre,
        year,
    } = data;

    return (
        <div className="poster-tile">
            <img className="poster-image" src={imageURL} alt="poster" />
            <div className="primary-data">
                <div className="poster-title">{title.toUpperCase()}</div>
                <div className="poster-year">{year}</div>
            </div>
            <div className="secondary-data">
                <div className="poster-genre">{genre}</div>
            </div>
        </div>
    );
};


export default PosterTile;
