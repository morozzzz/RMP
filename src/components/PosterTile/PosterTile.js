// @flow

import React from 'react';
import './PosterTile.css';

type posterTilePropType = {
    data: Object,
    onTileClick: Function
};

const PosterTile = ({ data, onTileClick }: posterTilePropType) => {
    const {
        poster_path,
        title,
        genres,
        release_date,
    } = data;

    const handleClick = () => {
        const selection = window.getSelection();
        const selectionType = selection && selection.type;

        if (selectionType !== 'Range') {
            onTileClick(data);
        }
    };

    return (
        <div onClick={handleClick} className="poster-tile">
            <img className="poster-tile__image" src={poster_path} alt="poster" />
            <div className="poster-tile__primary-data">
                <div className="poster-tile__title">{title.toUpperCase()}</div>
                <div className="poster-tile__year">{release_date.slice(0, 4)}</div>
            </div>
            <div className="poster-tile__secondary-data">
                <div className="poster-tile__genre">{genres.join(' ')}</div>
            </div>
        </div>
    );
};

export default PosterTile;
