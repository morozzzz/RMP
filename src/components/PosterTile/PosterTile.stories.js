import React from 'react';
import { storiesOf } from '@storybook/react';
import PosterTile from './PosterTile';

const data = {
    poster_path: 'https://image.tmdb.org/t/p/w500/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg',
    title: 'AVENGERS: INFINITY WAR',
    genres: ['Adventure', 'Science Fiction', 'Fantasy', 'Action'],
    release_date: '2018',
};

storiesOf('Poster Tile', module)
    .add('Default', () => (
        <div style={{ width: '300px' }}>
            <PosterTile data={data} />
        </div>
    ));
