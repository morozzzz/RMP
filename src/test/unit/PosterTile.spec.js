import React from 'react';
import { shallow } from 'enzyme';
import PosterTile from '../../components/PosterTile/PosterTile';

let selectionType;

const movieData = {
    title: 'Some title',
    release_date: '2002',
    poster_path: 'http://some_url',
    genres: [
        'Action',
        'Adventure',
    ],
};

const onTileClickMock = jest.fn();

global.getSelection = () => ({
    type: selectionType,
});

describe('PosterTile', () => {
    beforeEach(() => {
        onTileClickMock.mockClear();
    });

    it('should be rendered correctly', () => {
        const component = shallow(<PosterTile data={movieData} onTileClick={onTileClickMock} />);

        expect(component).toMatchSnapshot();
    });

    it('click on poster tile should invoke props.onTileClick() with right args', () => {
        const component = shallow(<PosterTile data={movieData} onTileClick={onTileClickMock} />);

        selectionType = 'None';

        component.simulate('click');

        expect(onTileClickMock).toHaveBeenCalledWith(movieData);
    });

    it('selection on poster tile should not invoke props.onTileClick()', () => {
        const component = shallow(<PosterTile data={movieData} onTileClick={onTileClickMock} />);

        selectionType = 'Range';

        component.simulate('click');

        expect(onTileClickMock).not.toBeCalled();
    });
});
