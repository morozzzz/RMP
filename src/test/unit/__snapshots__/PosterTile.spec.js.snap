// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosterTile should be rendered correctly 1`] = `
<div
  className="poster-tile"
  onClick={[Function]}
>
  <img
    alt="poster"
    className="poster-tile__image"
    src="http://some_url"
  />
  <div
    className="poster-tile__primary-data"
  >
    <div
      className="poster-tile__title"
    >
      SOME TITLE
    </div>
    <div
      className="poster-tile__year"
    >
      2002
    </div>
  </div>
  <div
    className="poster-tile__secondary-data"
  >
    <div
      className="poster-tile__genre"
    >
      Action Adventure
    </div>
  </div>
</div>
`;
